feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/following/FollowingFragment.java

```diff
@@ -1,33 +1,176 @@
 import android.view.ViewGroup;
-
+import android.widget.Toast;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
 import androidx.lifecycle.Observer;
 import androidx.lifecycle.ViewModelProviders;
-
+import com.baoyz.swipemenulistview.SwipeMenuListView;
+import com.example.feelslikemonday.DAO.FollowPermissionCallback;
+import com.example.feelslikemonday.DAO.FollowPermissionDAO;
+import com.example.feelslikemonday.DAO.UserCallback;
+import com.example.feelslikemonday.DAO.UserDAO;
+import com.example.feelslikemonday.DAO.VoidCallback;
 import com.example.feelslikemonday.R;
+import com.example.feelslikemonday.model.FollowPermission;
+import com.example.feelslikemonday.model.MoodEvent;
+import com.example.feelslikemonday.model.User;
+import com.example.feelslikemonday.model.followeeMoodEvent;
+import com.example.feelslikemonday.service.SortObjectDateTime;
+import com.example.feelslikemonday.ui.home.DisplayCurrentMood;
+import com.example.feelslikemonday.ui.login.SignupActivity;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import com.baoyz.swipemenulistview.SwipeMenu;
+import com.baoyz.swipemenulistview.SwipeMenuCreator;
+import com.baoyz.swipemenulistview.SwipeMenuItem;
+
+/**
+ * Responsible for listing all followees' moods and related information
+ */
 
-/*Responsible for listing all followers' moods and related information */
 public class FollowingFragment extends Fragment {
 
     private FollowingViewModel followingViewModel;
+    private SwipeMenuListView userList;
+    private FolloweeBookAdapter adapter;
+    private List<String> followeeUsernames;
+    private static FollowPermissionDAO DAO;
+    private SharedPreferences pref;
+    private String myUserID;
+    private List<MoodEvent> followeeUserMoodList;
+    private ArrayList<followeeMoodEvent> myfolloweeList = new ArrayList<>();
+    private int xIteration;
+    private int userVisited = 0;
+    private String  userCurrent;
 
-    public View onCreateView(@NonNull LayoutInflater inflater,
-                             ViewGroup container, Bundle savedInstanceState) {
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         followingViewModel = ViewModelProviders.of(this).get(FollowingViewModel.class);
         View root = inflater.inflate(R.layout.fragment_following, container, false);
         followingViewModel.getText().observe(getViewLifecycleOwner(), new Observer<String>() {
+
             @Override
             public void onChanged(@Nullable String s) {
-                //do stuff here
+
+                pref = getActivity().getApplicationContext().getSharedPreferences(SignupActivity.PREFS_NAME, 0);
+                myUserID = pref.getString(SignupActivity.USERNAME_KEY,null);
+                userList = getView().findViewById(R.id.followee_list);
+
+                DAO = FollowPermissionDAO.getInstance();

```

Comment:
We can set the dao when we create the object, as we do not need to get an instance every time we call this function

Reply:
but don't you want to get the new instance of the DAO every time because you want to get any changes done to the DAO

Reply:
No, the DAO is just an object that allows for DB transactions. Setting objects or changing objects through the DAO doesn't change the DAO itself

Reply:
ok i did it